manifests/dconf.pp

```diff
@@ -13,22 +13,44 @@
+  Stdlib::Absolutepath $base         = '/etc/dconf',
+  Optional[String]     $banner       = undef
 ) {
+  $_banner = $banner ? {
+    String  => $banner,
+    default => @(EOF)
+      --------------------------------- ATTENTION ----------------------------------
 
-  # Create an array of directories based on profile names
-  $locks_dir = regsubst($profile_list, '^(.*)$', '/etc/dconf/db/\1.d/locks')
-  $dir       = regsubst($profile_list, '^(.*)$', '/etc/dconf/db/\1.d')
+                               THIS IS A RESTRICTED COMPUTER SYSTEM
 
-  file { $dir :
-    ensure => 'directory',
-    owner  => 'root',
-    group  => 'root',
-    mode   => '0644'
+      This computer system, and all related equipment, networks, and
+      network devices are provided for authorised use only.  All
+      systems controlled by this organisation will be monitored for
+      all lawful purposes.  Monitoring includes the totality of the
+      operating system and connected networks.No events on this
+      system are excluded from record and there are no exclusions
+      from this policy.
+
+      Use of this system constitutes consent to full monitoring of
+      your activities for use by the authorised monitoring organisation.
+      Unauthorised use of this system, including uninvited connections,
+      may subject you to criminal prosecution.
+
+      The data collected from this system may be used for any purpose by
+      the collecting organisation.  If you do not agree to this
+      monitoring, discontinue use of the system IMMEDIATELY.
+
+                               THIS IS A RESTRICTED COMPUTER SYSTEM
+
+      --------------------------------- ATTENTION ----------------------------------
+      |EOF
   }
-  file { $locks_dir :
+
+  # Create an array of directories based on profile names
+  $_dir       = $profile_list.map |$profile| { "/etc/dconf/db/${profile}.d" }
+  $_locks_dir = $profile_list.map |$profile| { "/etc/dconf/db/${profile}.d/locks" }
+
+  file { $_dir + $_locks_dir :
```

Comment:
/etc/dconf/db is provided by the dconf rpm. We need to ensure it is installed before ensuring this file (and the one below it)